### PR TITLE
Feature: ERC20 approve

### LINKED ISSUE #809: ERC20 approve transactions

### PR DESCRIPTION
**Pull request by estebanmino**
**Description**

- This PR adds the ability to render adhoc information for erc20 approve transactions showing here https://recordit.co/IGgdpGmTR5
- Now we render the token information in transaction review
image
- Made some changes to avoid doing math calculation on render methods.
- Also we have a small bug with a missing locale `transaction.optional`

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

**Issue**

Resolves #809

### CODE REVIEW

File: locales/en.json [hunk 1]
```diff
@@ -461,7 +461,8 @@
 		"cancel_tx_title": "Attempt to cancel?",
 		"cancel_tx_message": "Submitting this attempt does not guarantee your original transaction will be cancelled. If the canellation attempt is successful, you will be charged the transaction fee above.",
 		"nevermind": "Nevermind",
-		"gasCancelFee": "Gas Cancellation Fee"
+		"gasCancelFee": "Gas Cancellation Fee",
+		"approve_warning": "By approving this action, you grant permission for this contract to spend up to"
```
Comment (brunobar79) [inline locales/en.json]: you will grant?

File: locales/es.json [hunk 2]
```diff
@@ -453,7 +453,8 @@
 		"value_not_available": "No disponible",
 		"rate_not_available": "Conversión no disponible",
 		"optional": "Opcional",
-		"no_address_for_ens": "No existe dirección para nombre ENS"
+		"no_address_for_ens": "No existe dirección para nombre ENS",
+		"approve_warning": "Al aprobar esta acción, otorgas permiso para que este contrato gaste hasta"
```
Comment (brunobar79) [inline locales/es.json]: otorgarás

**Review by brunobar79 — APPROVED** [review]
💯

File: locales/en.json (hunk 1 above)
Comment (estebanmino) [reply locales/en.json]: this is what we're saying on the extension 🤷‍♂

**Comment by ibrahimtaveras00** [comment]
Perhaps make the warning text align left instead of justified 

Smaller screen view:

image

**Review by ibrahimtaveras00 — APPROVED** [review]
Looks good now, QA Passed 👍

**Pull request merged**